### ISSUE TITLE
Idea: Treat Throwable->getTraceAsString() and __toString() as taint source for secrets

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#3696]
And potentially html and text, since params in the trace can be anything and are unescaped.

```php
<?php

function login($username, $password) {
    throw new RuntimeException('login failure');
}
try {
    login('user', 'pass');
} catch (Exception $e) {
    // This output includes 'pass'
    echo $e, "\n";
    echo $e->getTraceAsString();
}
```

**Comment by TysonAndre** [comment]
https://psalm.dev/r/73debb510a
Expected: Emit TaintedOutput
Observed: No issues

**Comment by TysonAndre** [comment]
If https://github.com/vimeo/psalm/issues/3709 is resolved, the following patch may help.

Also, I'm having issues combining `@psalm-taint-specialize` on a class with `psalm-taint-source` on a method - no warnings are emitted

```patch
diff --git a/src/Psalm/Internal/Stubs/CoreImmutableClasses.phpstub b/src/Psalm/Internal/Stubs/CoreImmutableClasses.phpstub
index 74e499efa..8283af3f3 100644
--- a/src/Psalm/Internal/Stubs/CoreImmutableClasses.phpstub
+++ b/src/Psalm/Internal/Stubs/CoreImmutableClasses.phpstub
@@ -16,13 +16,11 @@ class DateTimeZone
     public function __construct(string $tz) {}
 }
 
-/**
- * @psalm-taint-specialize
- */
 interface Throwable
 {
     /**
      * @psalm-mutation-free
+     * @psalm-taint-specialize
      */
     public function getMessage() : string;
 
@@ -30,6 +28,7 @@ interface Throwable
      * @psalm-mutation-free
      *
      * @return int|string https://www.php.net/manual/en/throwable.getcode.php
+     * @psalm-taint-specialize
      */
     public function getCode();
 
@@ -50,22 +49,28 @@ interface Throwable
 
     /**
      * @psalm-mutation-free
+     * @psalm-taint-specialize
      */
     public function getPrevious() : ?Throwable;
 
     /**
      * @psalm-mutation-free
+     * @psalm-taint-source input
      */
     public function getTraceAsString() : string;
+
+    /**
+     * @psalm-mutation-free
+     * @psalm-taint-source input
+     */
+    public function __toString() : string;
 }
 
-/**
- * @psalm-taint-specialize
- */
 class Exception implements Throwable
 {
     /**
      * @var string
+     * @psalm-taint-specialize
      */
     protected $message = '';
 
@@ -94,6 +99,7 @@ class Exception implements Throwable
 
     /**
      * @psalm-mutation-free
+     * @psalm-taint-specialize
      */
     public final function getMessage() : string {}
 
@@ -101,6 +107,7 @@ class Exception implements Throwable
      * @psalm-mutation-free
      *
      * @return int|string https://www.php.net/manual/en/throwable.getcode.php
+     * @psalm-taint-specialize
      */
     public final function getCode() {}
 
@@ -121,18 +128,23 @@ class Exception implements Throwable
 
     /**
      * @psalm-mutation-free
+     * @psalm-taint-specialize
      */
     public final function getPrevious() : ?Throwable {}
 
     /**
      * @psalm-mutation-free
+     * @psalm-taint-source input
      */
     public final function getTraceAsString() : string {}
+
+    /**
+     * @psalm-mutation-free
+     * @psalm-taint-source input
+     */
+    public final function __toString() : string {}
 }
 
-/**
- * @psalm-taint-specialize
- */
 class Error implements Throwable
 {
     /**
@@ -165,6 +177,7 @@ class Error implements Throwable
 
     /**
      * @psalm-mutation-free
+     * @psalm-taint-specialize
      */
     public final function getMessage() : string {}
 
@@ -190,11 +203,19 @@ class Error implements Throwable
 
     /**
      * @psalm-mutation-free
+     * @psalm-taint-specialize
      */
     public final function getPrevious() : ?Throwable {}
 
     /**
      * @psalm-mutation-free
+     * @psalm-taint-source input
      */
     public final function getTraceAsString() : string {}
+
+    /**
+     * @psalm-mutation-free
+     * @psalm-taint-source input
+     */
+    public final function __toString() : string {}
 }
```
